spec/rubocop/cop/style/hash_each_methods_spec.rb

```diff
@@ -144,6 +144,23 @@
         RUBY
       end
 
+      it 'registers an offense and corrects when `{hash: :literal}.keys.each`' do
+        expect_offense(<<~RUBY)
+          {hash: :literal}.keys.each { |k| p k }
+                           ^^^^^^^^^ Use `each_key` instead of `keys.each`.
+        RUBY
+
+        expect_correction(<<~RUBY)
+          {hash: :literal}.each_key { |k| p k }
+        RUBY
+      end
+
+      it "does not register an offense when `['array', 'literal'].keys.each`" do
+        expect_no_offenses(<<~RUBY)
+          ['array', 'literal'].keys.each { |k| p k }
```

Comment:
This code is broken.

```ruby
2.7.6 :002 > ['array', 'literal'].keys.each { |k| p k }
Traceback (most recent call last):
        4: from /home/ydakuka/.rvm/rubies/ruby-2.7.6/bin/irb:23:in `<main>'
        3: from /home/ydakuka/.rvm/rubies/ruby-2.7.6/bin/irb:23:in `load'
        2: from /home/ydakuka/.rvm/rubies/ruby-2.7.6/lib/ruby/gems/2.7.0/gems/irb-1.2.6/exe/irb:11:in `<top (required)>'
        1: from (irb):1
NoMethodError (undefined method `keys' for ["array", "literal"]:Array)
```

Reply:
The issue describes another case: when two args are passed to the `#each` and second one is not used.
```ruby
[[1, 2, 3], [4, 5, 6]].each { |a, _| puts(a) }
```